src/worker.js

```diff
@@ -19,19 +19,18 @@ clsBlueBird(cls)
 
 const optsSchema = joi.object({
   attempt: joi.number().integer().min(0).required(),
-  done: joi.func().required(),
   errorCat: joi.object(),
   finalRetryFn: joi.func(),
   jobSchema: joi.object({
     isJoi: joi.bool().valid(true)
   }).unknown(),
   job: joi.object().required(),
   log: joi.object().required(),
-  maxNumRetries: joi.number().integer().min(0),
-  msTimeout: joi.number().integer().min(0),
+  maxNumRetries: joi.number().integer().min(0).required(),
+  msTimeout: joi.number().integer().min(0).required(),
```

Comment:
this is not required for the user, it is required in the constructor. We default these